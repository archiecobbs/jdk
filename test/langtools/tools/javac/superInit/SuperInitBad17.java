/*
 * @test /nodynamiccopyright/
 * @bug 8193760
 * @summary Verify non-canonical record constructors only invoke this()
 *
 * @compile/fail/ref=SuperInitBad17.out -XDrawDiagnostics SuperInitBad17.java
 */
class SuperInitBad17 {
    record Record(int value) {
        Record(float x) {
            if (x < 0)
                this((int)x);
            else
                super();
        }
    }
}
