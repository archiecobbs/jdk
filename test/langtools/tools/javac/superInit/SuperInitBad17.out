SuperInitBad17.java:10:9: compiler.err.non.canonical.constructor.invoke.other.constructors: SuperInitBad17.Record
1 error
